File: cineticket-ui/src/types/movie.ts
export interface Movie {
  id: number;
  title: string;
  description: string;
  poster_url: string;
  duration_min: number;
  // Backend'den gelen diğer alanlar (created_at vs.) şimdilik lazım değil
}

export interface Showtime {
  id: number;
  movie_id: number;
  start_time: string; // Backend'den tarih string gelir (ISO 8601)
  price: number;
}

File: cineticket-api/internal/models/movie.go
package models

import "time"

type Movie struct {
	ID          int64     `db:"id" json:"id"`
	Title       string    `db:"title" json:"title"`
	Description string    `db:"description" json:"description"`
	PosterURL   string    `db:"poster_url" json:"poster_url"`
	DurationMin int       `db:"duration_min" json:"duration_min"`
	CreatedAt   time.Time `db:"created_at" json:"created_at"`
}
